src/intrprtr.c

```diff
@@ -47,58 +47,21 @@
 #include "hpc/guards.h"
 #endif
 
-/****************************************************************************
-**
-*V  IntrReturning   . . . . . . . . . . .  interpreter is currently returning
-**
-**  If 'IntrReturning' is  non-zero, the interpreter is currently  returning.
-**  The interpreter switches  to this mode when  it finds a return-statement.
-**  If it interprets a return-value-statement, it sets 'IntrReturning' to 1.
-**  If it interprets a return-void-statement,  it sets 'IntrReturning' to 2.
-**  If it interprets a quit-statement, it sets 'IntrReturning' to 8.
-*/
-/* TL: UInt IntrReturning; */
-
-
-/****************************************************************************
-**
-*V  IntrIgnoring  . . . . . . . . . interpreter is currently ignoring actions
-**
-**  If 'IntrIgnoring'  is  non-zero,  the interpreter  is  currently ignoring
-**  actions.  The interpreter switches to this mode for  the right operand of
-**  'or' and 'and'  constructs where the  left operand already determines the
-**  outcome.
-**
-**  This mode is also used in Info and Assert, when arguments are not printed.
-*/
-/* TL: UInt IntrIgnoring; */
-
-
-/****************************************************************************
-**
-*V  IntrCoding  . . . . . . . . . . . interpreter is currently coding actions
-**
-**  If 'IntrCoding' is non-zero, the interpreter is currently coding actions.
-**  The interpreter  switches  to this  mode for  constructs  that it  cannot
-**  directly interpret, such as loops or function bodies.
-*/
-/* TL: UInt IntrCoding; */
-
 // INTERPRETER_PROFILE_HOOK deals with profiling of immediately executed
 // code.
-// If STATE(IntrCoding) is true, profiling is handled by the AST
+// If intr->IntrCoding is true, profiling is handled by the AST
 // generation and execution. Otherwise, we always mark the line as
-// read, and mark as executed if STATE(IntrReturning) and STATE(IntrIgnoring)
+// read, and mark as executed if intr->IntrReturning and intr->IntrIgnoring
 // are both false.
 //
 // IgnoreLevel gives the highest value of IntrIgnoring which means this
 // statement is NOT ignored (this is usually, but not always, 0)
-static void INTERPRETER_PROFILE_HOOK(int ignoreLevel)
+static void INTERPRETER_PROFILE_HOOK(IntrState * intr, int ignoreLevel)
```

Comment:
The majority of the changes below either add an `IntrState * intr` argument to functions; or pass `intr` on to other functions that need them; or replace `STATE(IntrFoo)` by `intr->foo`.

From here on, a long time nothing else happens... I think for the rest of the whole file `intrprtr.c`